pennylane/devices/default_mixed.py

```diff
@@ -335,6 +335,50 @@ def _apply_channel(self, kraus, wires):
 
         self._state = qnp.einsum(einsum_indices, kraus, self._state, kraus_dagger)
 
+    def _apply_channel_tensordot(self, kraus, wires):
+        r"""Apply a quantum channel specified by a list of Kraus operators to subsystems of the
+        quantum state. For a unitary gate, there is a single Kraus operator.
+
+        Args:
+            kraus (list[array]): Kraus operators
+            wires (Wires): target wires
+        """
+        channel_wires = self.map_wires(wires)
+        num_ch_wires = len(channel_wires)
+
+        # Shape kraus operators and cast them to complex data type
+        kraus_shape = [2] * (num_ch_wires * 2)
+        kraus = [qnp.cast(qnp.reshape(k, kraus_shape), dtype=self.C_DTYPE) for k in kraus]
+
+        # row indices of the quantum state affected by this operation
+        row_wires_list = channel_wires.tolist()
+        # column indices are shifted by the number of wires
+        col_wires_list = [w + self.num_wires for w in row_wires_list]
+
+        channel_col_ids = list(range(num_ch_wires, 2 * num_ch_wires))
+        axes_left = [channel_col_ids, row_wires_list]
+        # Use column indices instead or rows to incorporate transposition of K^\dagger
+        axes_right = [col_wires_list, channel_col_ids]
+        # Apply the Kraus operators, and sum over all Kraus operators afterwards
+        _state = qnp.sum(
+            qnp.stack(
```

Comment:
Is stacking bringing some performance boost?

Reply:
Haha, this is an excellent pointer, thank you!
`stack` is not only _not_ giving a boost, it is actually part of what makes `_apply_channel_tensordot` slower than `_apply_channel` ! If I skip the `stack`, the `tensordot`-based method is (considerably!) faster even for a single-qubit operation (compare the timings [here](https://github.com/PennyLaneAI/pennylane/pull/3584/files/d5bcd875ea0bfe75864e2dcf5c505156e33fb99c#r1064387589) )
![Screenshot from 2023-01-09 09-54-49](https://user-images.githubusercontent.com/20772922/211270808-5b73756a-4a62-4340-89fd-053fd6c73842.png)

However, without `stack`, we loose compatibility with all of the interfaces. It would be great if this would be fixable without stacking. Do you have an idea?

Reply:
I couldn't find a solution yet, but have the following proposal:
We could branch off the unitary case `len(kraus)==1`, and skip the stacking -- and indeed also the `sum` call -- in that case. This would allow us to use the faster `stack`-free variant for unitary gates.
In this case, it might actually be interesting to replace `_apply_channel` entirely, as you suggested.

Reply:
I like your proposal, I think it is a good idea and allow us to reduce the amount of code to maintain :100:

Reply:
Not going for this in the end, because we actually need to maintain _apply_channel and _apply_channel_tensordot, and additional branching probably is not worth it.
resolving and looking forward to the new device API :rocket: